api-tests/src/steps/ipv-steps.ts

```diff
@@ -150,6 +154,84 @@ Then(
   },
 );
 
+When("I return using my identity", async function (this: World): Promise<void> {
+  this.ipvSessionId = await internalClient.initialiseIpvSession(
+    await generateInitialiseIpvSessionBody(
+      this.userId,
+      this.journeyId,
+      this.journeyType,
+    ),
+  );
+  this.lastJourneyEngineResponse = await internalClient.sendJourneyEvent(
+    "/journey/next",
+    this.ipvSessionId,
+  );
+});
+
+Then(
+  "My identity {string} is {string}",
+  function (this: World, field: string, value: string): void {
+    let valueToMatch;
+    switch (field) {
+      case "GivenName":
+      case "FamilyName": {
+        const namePart = this.identity[CORE_IDENTITY].name?.[0].nameParts.find(
+          (np) => {
+            return np.type === field;
+          },
+        );
+        valueToMatch = namePart?.value?.toLowerCase();
+        break;
+      }
+      case "buildingNumber": {
+        valueToMatch =
+          this.identity?.[ADDRESS_IDENTITY]?.[0].buildingNumber?.toLowerCase();
+        break;
+      }
+      case "addressCountry": {
+        valueToMatch =
+          this.identity?.[ADDRESS_IDENTITY]?.[0].addressCountry?.toLowerCase();
+        break;
+      }
+      case "uprn": {
+        valueToMatch = this.identity?.[ADDRESS_IDENTITY]?.[0].uprn;
+        break;
+      }
+      case "streetName": {
+        valueToMatch =
+          this.identity?.[ADDRESS_IDENTITY]?.[0].streetName?.toLowerCase();
+        break;
+      }
+      case "postalCode": {
+        valueToMatch =
+          this.identity?.[ADDRESS_IDENTITY]?.[0].postalCode?.toLowerCase();
+        break;
+      }
+      case "addressLocality": {
+        valueToMatch =
+          this.identity?.[ADDRESS_IDENTITY]?.[0].addressLocality?.toLowerCase();
+        break;
+      }
+      case "validFrom": {
+        valueToMatch =
+          this.identity?.[ADDRESS_IDENTITY]?.[0].validFrom?.toLowerCase();
+        break;
+      }
+      case "buildingName": {
+        valueToMatch =
+          this.identity?.[ADDRESS_IDENTITY]?.[0].buildingName?.toLowerCase();
+        break;
+      }
+      case "subBuildingName": {
+        valueToMatch =
+          this.identity?.[ADDRESS_IDENTITY]?.[0].subBuildingName?.toLowerCase();
+        break;
```

Comment:
You could simplify this by creating separate steps for address and name, e.g.:
```
Then(
  "my address {string} is {string}",
  function (this: World, field: keyof PostalAddressClass, value: string): void {
    assert.equal(
      value.toLowerCase(),
      this.identity?.[ADDRESS_IDENTITY]?.[0][field]?.toString().toLowerCase(),
    );
  },
);
```
(I think it should also be lowercase 'my')